dataactbroker/handlers/fileHandler.py

```diff
@@ -779,6 +780,75 @@ def check_generation(submission, file_type):
 
         return JsonResponse.create(StatusCode.OK, response_dict)
 
+    @staticmethod
+    def submit_detached_file(submission):
+        """ Submits the FABS upload file associated with the submission ID """
+        # Check to make sure it's a d2 submission
+        if not submission.d2_submission:
+            raise ResponseException("This is not a FABS submission", StatusCode.CLIENT_ERROR)
+
+        # Check to make sure it isn't already a published submission
+        if submission.publish_status_id != PUBLISH_STATUS_DICT['unpublished']:
+            raise ResponseException("This submission has already been published", StatusCode.CLIENT_ERROR)
+
+        # if it's an unpublished FABS submission, we can start the process
+        sess = GlobalDB.db().session
+        submission_id = submission.submission_id
+
+        try:
+            # get all valid lines for this submission
+            query = sess.query(DetachedAwardFinancialAssistance).\
+                filter_by(is_valid=True, submission_id=submission_id).all()
+
+            for row in query:
+                # if it is not a delete row
+                if row.correction_late_delete_ind is None or row.correction_late_delete_ind.upper() != "D":
+                    # remove all keys in the row that are not in the intermediate table
+                    temp_obj = row.__dict__
+                    temp_obj.pop('detached_award_financial_assistance_id', None)
```

Comment:
Not that I've seen. I could make a for loop and a list of all the keys I want out but all that would do would be to make it look nicer. It wouldn't actually change how it's working.